tools/clang/tools/dxclib/dxc.cpp

```diff
@@ -1216,6 +1218,36 @@ void DxcContext::GetCompilerVersionInfo(llvm::raw_string_ostream &OS) {
 #define VERSION_STRING_SUFFIX ""
 #endif
 
+#ifdef _WIN32
+// Unhandled exception filter called when an unhandled exception occurs
+// to at least print an generic error message instead of crashing silently.
+// passes exception along to allow crash dumps to be generated
+static LONG CALLBACK ExceptionFilter(PEXCEPTION_POINTERS pExceptionInfo)
+{
+  if (pExceptionInfo && pExceptionInfo->ExceptionRecord) {
+    switch(pExceptionInfo->ExceptionRecord->ExceptionCode) {
+    case EXCEPTION_ACCESS_VIOLATION: {
+      const char *readWrite;
+      if (pExceptionInfo->ExceptionRecord->ExceptionInformation[0])
+        readWrite = "write";
+      else
+        readWrite = "read";
+      std::cerr <<
+        "Internal compiler error: access violation. Attempt to " << readWrite <<
+        " from address 0x" << std::hex << std::setfill('0') << std::setw(16) <<
+        pExceptionInfo->ExceptionRecord->ExceptionInformation[1] << std::endl;
+    } break;
+    case EXCEPTION_STACK_OVERFLOW:
+      std::cerr << "Internal compiler error: stack overflow" << std::endl;
+    }
```

Comment:
We should have a default print for unhandled exception here.

Reply:
We also have the decompile-specific exceptions (like we throw for assets). Would be good to add those here too